### ISSUE TITLE
Season-indexed occupancy state (spOccupancy tPGOcc) + season-varying rates on dyn_occu

### ISSUE THREAD

**Issue by gcol33** [gcol33/tulpaObs#124]
## The gap

From the 2026-07-16 audit against `spOccupancy` / `spAbundance`. This one is not
a missing *function* but a missing *model*: `dyn_occu()` and `tPGOcc` are
different models, and the roster currently reads as though one covers the other.

**`dyn_occu()` is colext.** `src/dyn_occ_likelihood.h`:

```
// Process 0: initial occupancy -- eta[0] = logit(psi1_i)
// Process 2: colonization      -- eta[2] = logit(gamma_i)
// Process 3: extinction        -- eta[3] = logit(epsilon_i)
// alpha[t] = alpha[t-1]*(1-epsilon) + (1-alpha[t-1])*gamma
// Reference: MacKenzie et al. (2003) Ecology
```

**`tPGOcc` is not.** It models multi-season occupancy with covariates and
temporal random effects (`ar1 = TRUE`) on the state -- no explicit gamma/epsilon.
Both are legitimate; neither substitutes for the other. A user who wants "occupancy
trend over 9 years with an AR1 year effect" has no route today.

## What the `temporal()` route actually gives (and why it is not tPGOcc)

`temporal()` composes with the areal field under `nested_laplace`, but the field
is indexed **per site, not per season** (`.tobs_block_from_temporal`,
R/em_nested_laplace.R):

```r
time_idx_site <- as.integer(temporal$time_idx)
if (length(time_idx_site) != n_sites) stop(...)
```

One time index per row of site-level `data`. The only fixture
(`simulate_panel_occu`) does `time_idx <- sample(seq_len(n_times), n_sites,
replace = TRUE)` -- one timestamp per site. So the tPGOcc-analogue you can build
is a stacked site-year panel with **independent z per row** and an AR1 field over
years. There is no `[site x season]` occupancy state with AR1 on the state.

Worse for `dyn_occu`: a `temporal()` term there enters **psi1 only**
(`R/laplace_callbacks.R:303`, "the state field enters season-1 occupancy psi1
only"), i.e. it is a season-1 field, not a season-indexed one.

## Three sub-gaps, ranked

- [ ] **Season-indexed occupancy state (the tPGOcc model).** A `[site x season]`
      latent z with a temporal field/RE on the state and no colonization/extinction
      parameterization. This is a new marginal, not a re-wiring: with AR1 on the
      state the seasons do not factorize, so it is an HMM-with-continuous-state or
      an AGHQ/Laplace integration over the year effect.
- [ ] **Season-varying colonization / extinction covariates on `dyn_occu`.**
      `logit_gam = eta[2]` / `logit_eps = eta[3]` are read once per site, outside
      the season loop, never indexed by `t`. `dyn_abun` already solved exactly this
      (#80): a covariate carried as an `[n_sites x (T-1)]` matrix column drives
      interval-indexed vital rates, with per-interval forward-mode gradients and a
      scalar overload that is byte-identical to the constant-rate path. That recipe
      ports directly to the `dyn_occ` forward kernel.
- [ ] **Season-varying detection / occupancy covariates on `dyn_occu`.** All
      designs are site-level (`build_dynamic_callbacks`: "X_det is site-indexed in
      this model, so per-season rows just replicate the site's covariates"). Note
      `dyn_occ_likelihood.h` reads `dyn->X_det_visit` when `p_det_visit > 0`, but
      `src/occu_fit.cpp` never populates it on the dynamic branch
      (`spec.n_extra_params = 0`) and `R/occu_fit.R` only sets `spec$X_det_visit`
      on `model_type == "single"` -- so that branch is **dead code**. Either wire it
      or delete it; leaving it reads as support that does not exist.

The middle item is the cheapest and has a shipped template (#80). The first is the
real family.

## Definition of Done

Per item: recovery vs simulated truth across >= 20 seeds, 95% coverage >= 0.85.
The season-varying items get the `dyn_abun` #80 anchor -- a constant-covariate
fixture must be byte-identical to the current constant-rate path.

Related: #114 (structured-term breadth on observation families), #80 (the
season-varying template in `dyn_abun`).

**Comment by gcol33** [comment]
Closed by 0.0.146–0.0.151. All three ranked sub-gaps addressed, each with >= 20-seed recovery:

**1. Season-indexed occupancy state (the tPGOcc model) — `t_occu()` (0.0.151).**
A `[site x season]` Bernoulli occupancy `z_{i,t} ~ Bern(psi_{i,t})`, `logit psi = X beta + eta_t`, with a shared AR1 year effect `eta_t = rho eta_{t-1} + w_t` on the logit and no colonization/extinction. Rather than a marginal Laplace/AGHQ integration over the year effect, this is fit by the exact Polya-Gamma Gibbs sampler spOccupancy actually uses for `tPGOcc`: conditional on the year effects the seasons factorise, so a sweep draws z, then the joint `(beta_occ, eta)` as one GMRF update with the stationary AR1 precision as the year-effect prior (exactly as spPGOcc's ICAR precision is for a spatial field), then `beta_p`, then the AR1 hyperparameters (`sigma^2` Inverse-Gamma, `rho` on a grid). The AR1 hypers update on the raw year series before the sum-to-zero centering (else the centering distorts the increments and `rho` collapses). `R/t_occu.R`, `simulate_t_occu()`, `test-t-occu.R`.

The year-effect surface (`fit$temporal_field`, cor ~0.97), the occupancy/detection coefficients, and the innovation SD `sigma` recover with nominal coverage (`Rhat` ~1.00). The AR1 **correlation** `rho` is a weakly-identified parameter of a short time series — given even the true year effects `rho_hat` climbs ~0.08 (T=8) -> 0.43 (T=20) -> 0.59 (T=200) for truth 0.6 (`dev_notes/_probe_tpg_rho.R`), an identifiability property of the short AR1, not the sampler — so it is reported but not asserted tightly.

**2. Season-varying colonization/extinction on `dyn_occu` (0.0.148).**
The `dyn_abun` #80 recipe ported to the colext forward: a `[n_sites x (T-1)]` covariate column drives interval-indexed vital rates, per-interval forward-backward, the constant-covariate path byte-identical to the constant-rate path. `simulate_dyn_occu(beta_gamma=, beta_epsilon=)`, 20-seed `test-dyn-occ.R`.

**3. Season-varying detection on `dyn_occu` (0.0.148) + the `X_det_visit` dead code.**
Per-season detection via a `[n_sites x T]` covariate on the `detection` formula, sharing the period-agnostic unroller; 20-seed `test-dyn-occ.R`. The flagged dead read (`dyn_occ_likelihood.h` consulting `X_det_visit` when `p_det_visit > 0`, never populated on the dynamic branch) is **deleted** (fe30bc1) — the dynamic detection design is site-level, optionally season-indexed; the removed branch never executed so the NUTS forward is byte-identical.

DoD met: >= 20-seed recovery, single-covariate fixtures byte-identical to the constant paths. NUTS + season-varying and a within-season visit-level detection covariate remain clean separable follow-ups, not dead reads.
